src/mame/heathkit/h89.cpp

```diff
@@ -100,12 +102,15 @@ class h89_state : public driver_device
 	bool m_timer_intr_enabled;
 	bool m_floppy_ram_wp;
 
+	int m_cpu_speed_multiplier;
```

Comment:
If you don’t register `m_cpu_speed_multiplier` for save states (and change it to an explicitly sized integer), you’ll get strange behaviour.  The next time “fast” mode is switched on, the multiplier will come from the config setting on the most recent reset, not the reset before the saved state was created.